### DISCUSSION TITLE
How to update FRAME_BUFFER texture size dynamically

### DISCUSSION THREAD

**Discussion by FuXiii** [openscenegraph/OpenSceneGraph#1272] [Q&A] [answered]
I try to use `osg::camera::attach(...)` to implement `post-process`. All examples in `osg` , size of the texture attach for `COLOR_BUFFER` is fixed. If I resize my `Viewer` I need to resize the texture size which I had attached. If I directly resize it in `update` it doesn't work.

```c++
// in one of camera preDrawCallback
osg::ref_pte<osg::Texture2D> color_buffer;
void operator()(osg::RenderInfo& renderInfo) const override
{
      ...
      double current_viewer_width = ...//Get current viewer windows width
      double current_viewer_height = ...//Get current viewer windows height
      color_buffer->setTextureSize(current_viewer_width, current_viewer_height );

      // color_buffer->dirtyTextureObject(); // I try to dirty this texture, but it doesn't work.
}
```

How to resize the `COLOR_BUFFER` texture size dynamically?

**Comment by robertosfield** [comment]
I don't recall the details as it's over 5 years since I worked on the OSG, but GPUs don't like changing sizes of resources, they work best when you allocate a texture/framebuffer etc. at a fixed size for the life of the application.  If you want to change the size you need to reallocate that resource.  The is the same for OpenGL/OpenSceneGraph as it is for Vulkan/VulkanSceneGraph.

In the case of "resizing" textures the most efficient route is to allocate to the largest size you need then if you don't need to whole extents use a smaller a viewport to it, and adjust the texgen/texcoords when reference it.

**Reply by FuXiii** [reply]
Use a large FBO and just use smaller viewport, looks very interesting, good idea, I will try it. (｡•̀ᴗ-)✧

**Comment by xarray** [comment]
Have a look at osgverse (https://github.com/xarray/osgverse).
It supports FBO size updating when you resize the window. All work are done in struct MyResizedCallback, in pipeline/Pipeline.cpp.

**Reply by FuXiii** [reply]
Is this [Pipeline.cpp:550](https://github.com/xarray/osgverse/blob/12989733c3294669b5693551c5a1bb5289fbcc11/pipeline/Pipeline.cpp#L550C9-L550C9) line code resize the FBO size? I had not found any code about resize the `Texture` which attached in `RTT` camera, is that means just resize the `osg::GraphicsContext::Traits` will synchronous update the FBO size?

By the way, two Chinese speak English, which is a bit strange.✧*｡ (ˊᗜˋ*) ✧*

王锐大佬您好~哈哈

**Reply by xarray** [reply, marked as answer]
camera->resizeAttachments() should do the trick, as I remember.

PS: yes we can speak Chinese in wechat groups or somewhere else, but not suitable in a public issue. :)

**Reply by FuXiii** [reply]
I tried this method, it fixed my issue. Thanks~( ๑╹ ꇴ╹) ｸﾞｯ!
